src/averageAges.js

```diff
@@ -20,6 +20,14 @@ function calculateMenAverageAge(people, century) {
   // avoid using loop and forEach
   // replace `if ()` statement with &&, || or ?:
   // without nesting
+  const arrayWithMen = people.filter(el => el.sex === 'm');
+  const diedThisCentury = century
+    ? arrayWithMen.filter(el => Math.ceil(el.died / 100) === century)
+    : arrayWithMen;
+  const sumOfAges = diedThisCentury.reduce((sum, num) =>
+    sum + (num.died - num.born), 0);
+
+  return sumOfAges / diedThisCentury.length;
```

Comment:
would be great to move this logic to a separate function and reuse in next fuction